src/services/reviews.js

```
@@ -739,6 +770,142 @@ module.exports = class reviewsHelper {
+
+			return programRolloutId
+		} catch (error) {
+			throw new Error('Program Rollout Update failed. Error : ', error)
+		}
+	}
+
+	/**
+	 * Create Program Rollout
+	 * @method
+	 * @name createProgramRollout
+	 * @param {Object} programData - Program data object
+	 * @returns {Integer} - program rollout id
+	 */
+
+	static async createProgramRollout(programData) {
+		try {
+			// rollout request body
+			const rolloutReqBody = {
+				resource_id: programData?.id,
+				resource_type: programData?.type,
+				start_date: programData?.meta?.start_date,
+				end_date: programData?.meta?.end_date,
+				targeting_criteria: programData?.targeting_criteria,
+				title: programData.title,
+				resources: programData?.resources,
+			}
+
+			// create an entry to rollout table
+			const createProgramRollout = await rolloutService.create(
+				rolloutReqBody,
+				programData.user_id,
+				programData.organization_id
+			)
+
+			if (createProgramRollout.responseCode !== httpStatusCode.ok) {
+				throw new Error(`Rollout creation failed: ${createProgramRollout.message || 'Unknown error'}`) // Include error message if available
+			}
+
+			rolloutId = createProgramRollout?.result?.id
+
+			let createRolloutPromise = []
+
+			programData?.resources.forEach(async (resource) => {
+				const resourceRolloutReqBody = {
+					resource_id: resource?.id,
+					resource_type: resource?.type,
+					start_date: resource?.start_date || '',
+					end_date: resource?.end_date || '',
+					targeting_criteria: resource?.targeting_criteria,
+					title: resource.title,
+					resources: resource?.resources,
+				}
+				createRolloutPromise.push(
+					await rolloutService.create(
+						resourceRolloutReqBody,
+						resource.user_id,
+						resource.organization_id,
+						true
+					)
+				)
+			})
+
+			await Promise.all(createRolloutPromise)
+
+			return rolloutId
+		} catch (error) {
+			throw new Error('Program Rollout Creation failed. Error : ', error)
```

@adithyadinesh0412  here we need to return that error